src/renderer_mtl.mm

```diff
@@ -3492,14 +3523,20 @@ void writeString(bx::WriterI* _writer, const char* _str)
 
 		if (sampleCount > 1)
 		{
+#if !BX_PLATFORM_VISIONOS
```

Comment:
In cases where both #if/#else path can compile on both platforms, prefer using compiler instead preprocess to determine branch.

```cpp
desc.pixelFormat = BX_ENABLED(BX_PLATFORM_VISIONOS)
    ? MTLPixelFormatBGRA8Unorm_sRGB
    : m_metalLayer.pixelFormat
    ;
```